### DISCUSSION TITLE
Parring wallet ( GET WALLET ADDRESS)

### DISCUSSION THREAD

**Discussion by ciprian-macovei** [WalletConnect/walletconnect-monorepo#4050] [Developer Support]
hello guys , im using walletconnect, got the pairing right received the object, but how can I get the user wallet address from pairing ?
please help me if it is possible ill paste here the info object I receive after pairing .
<img width="429" alt="Screenshot 2023-12-16 at 12 14 25" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/49752796/7ff229aa-5e67-46c1-a3bb-70e189665039">

**Comment by glitch-txs** [comment]
hi could you provide more information?, which library and versions are you using and a code snippet? Thanks!

**Reply by ciprian-macovei** [reply]
import AuthClient from '@walletconnect/auth-client';
import { createWeb3Modal, defaultConfig } from '@web3modal/ethers5/react'


  "@walletconnect/auth-client": "^2.1.2",
    "@walletconnect/client": "^1.8.0",
    "@walletconnect/sign-client": "^2.10.6",
    "@web3modal/ethers5": "^3.5.0",
    
    
    so is there a way I can get wallet info from the URI generated by walletconnect,
    
    example: I go click connect then I say I want I copy the link
    
<img width="389" alt="Screenshot 2023-12-19 at 00 09 06" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/49752796/326fe000-3f41-4370-a6fe-d10fc175180a">

    
    is there a way I can obtain info about the account trying to do that,
    
    I know this is about paring, I have this code:
    
    

    if (symLink) {
                const res = await auth?.core.pairing.pair({ uri: symLink })
                if (res?.topic) {
                    const a = await auth?.core.pairing.activate({ topic: res.topic });

**Reply by glitch-txs** [reply]
auth client has been deprecated please don't use it, we have [SIWE](https://docs.walletconnect.com/web3modal/react/siwe/about) integration for auth
